content/200-concepts/100-components/01-prisma-schema/09-index-configuration.mdx

```diff
@@ -0,0 +1,143 @@
+---
+title: 'Index configuration'
+metaDescription: 'How to configure index functionality to allow sorting and length restriction'
+hidePage: false
+tocDepth: 2
+preview: true
+---
+
+<TopBlock>
+
+From version `3.5.0` onwards, Prisma will allow configuration of indexes, unique constraints and primary keys constraints using extended syntax options in the Prisma schema language. Two new arguments, `length` and `sort`, can now be specified in the `@id`, `@@id`, `@unique`, `@@unique` and `@@index` fields in certain databases:
+
+- The `length` argument is available on MySQL on the `@id`, `@@id`, `@unique`, `@@unique` and `@@index` fields. It allows Prisma to now support indexes and constraints on `String` and `Bytes` types.
+
+- The `sort` argument is available for all databases on the `@unique`, `@@unique` and `@@index` fields. Additionally, SQL Server also allows it on `@id` and `@@id`.
+
+<Admonition type="warning">
+
+This might be a **breaking change** if you have a database that is already making use of these features, so it needs to be explicitly enabled via a Preview feature. See [Upgrading from previous versions](#upgrading-from-previous-versions) for details.
+
+</Admonition>
+
+In order to enable this as a Preview feature please add it to the `generator` block:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["extendedIndexes"]
+}
+```
+
+</TopBlock>
+
+## New `length` argument (MySQL)
+
+The `length` argument is specific to MySQL and will allow you to define indexes and constraints on columns of `String` and `Byte` types. For these types, MySQL requires you to specify a maximum length for the subpart of the value to be indexed in cases where the full value would exceed MySQL's limits for index sizes. See [the MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/innodb-limits.html) for more details.
+
+The `length` argument is available on the `@id`, `@@id`, `@unique`, `@@unique` and `@@index` fields.
+
+As an example, the following data model declares a field with a max length of 3000 characters as id.
+
+```prisma file=schema.prisma
+model Id {
+  id String @id @db.VarChar(3000)
+}
+```
+
+But this is not not valid in MySQL because it exceeds MySQL's index storage limit and therefore Prisma rejects the data model. The generated SQL would be rejected by the database.
```

Comment:
Do we know the error message this would create in Prisma? Might be useful to add here.

Reply:
Thanks, going to mark this as resolved for now as we're going for a sort of 'minimal viable documentation' version to get out now, but this would be useful to have in an updated version if @do4gr is able to provide it